micropip/logging.py

```diff
@@ -87,17 +87,48 @@ def _set_formatter_once() -> None:
     _logger.addHandler(ch)
 
 
-def setup_logging(verbosity: int | bool) -> logging.Logger:
-    _set_formatter_once()
+class LoggerWrapper:
+    __slots__ = ("logger", "_orig_level")
+
+    def __init__(self, logger: logging.Logger):
+        self.logger = logger
+        self._orig_level = logger.level
+
+    def restore(self):
+        self.logger.setLevel(self._orig_level)
+
+    @contextmanager
+    def restore_temp(self):
+        cur_level = self.logger.level
+        self.logger.setLevel(self._orig_level)
+        try:
+            yield
+        finally:
+            self.logger.setLevel(cur_level)
+
 
+def setup_logging(verbosity: int | bool | None) -> LoggerWrapper:
+    _set_formatter_once()
+    assert _logger
+    result = LoggerWrapper(_logger)
```

Comment:
I think this is still wrong, as this changes the  logging level permanently for a logger wrapper even if passed None. 

I think the conflation of getting a logger and setting the level is creating issues.